src/ApplicationIO.hpp

```diff
@@ -39,34 +42,34 @@ namespace geopm
             /// @brief Returns the set of region names recorded by the
             ///        application.
             virtual std::set<std::string> region_name_set(void) const = 0;
-            /// @brief Signal to the application that the Controller
-            ///        is ready to begin receiving samples.
-            virtual void controller_ready(void) = 0;
-            /// @brief Signal to the application that the Controller
-            ///        has failed critically.
-            virtual void abort(void) = 0;
     };
 
-    class ApplicationSampler;
-
     class ApplicationIOImp : public ApplicationIO
     {
         public:
             ApplicationIOImp();
-            ApplicationIOImp(ApplicationSampler &application_sampler);
+            ApplicationIOImp(std::shared_ptr<ServiceProxy> service_proxy,
+                             const std::string &profile_name,
+                             const std::string &report_name,
+                             int timeout,
+                             int num_proc);
             virtual ~ApplicationIOImp();
-            void connect(void) override;
-            bool do_shutdown(void) const override;
+            std::vector<int> connect(void) override;
+            bool do_shutdown(void) override;
             std::string report_name(void) const override;
             std::string profile_name(void) const override;
             std::set<std::string> region_name_set(void) const override;
-            void controller_ready(void) override;
-            void abort(void) override;
         private:
+            std::set<int> get_profile_pids(void);
             static constexpr size_t M_SHMEM_REGION_SIZE = 2*1024*1024;
 
             bool m_is_connected;
-            ApplicationSampler &m_application_sampler;
+            std::shared_ptr<ServiceProxy> m_service_proxy;
+            const std::string m_profile_name;
+            const std::string m_report_name;
+            const int m_timeout;
```

Comment:
Nit: should these be uppercase?

Reply:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es9-avoid-all_caps-names

I have never been a fan of constant members being all caps, although I am not in the majority wrt contributors to the project.  Consistency is important, but there are a number of arguments against all caps members.